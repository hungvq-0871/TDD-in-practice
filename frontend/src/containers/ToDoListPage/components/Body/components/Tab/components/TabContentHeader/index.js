import React from 'react';
import PropTypes from 'prop-types';

import TabContentHeaderStyle from './index.style';

function TabContentHeader({
  dueToday,
  closeToday,
}) {
  return (
    <TabContentHeaderStyle>
      <span>
        Due Today (<span className="due-today-counter">{dueToday}</span>)
      </span>
      <span className="close-day">
        Close Today (<span className="close-today-counter">{closeToday}</span>)
      </span>
    </TabContentHeaderStyle>
  )
}

TabContentHeader.propTypes = {
  dueToday: PropTypes.number.isRequired,
  closeToday: PropTypes.number.isRequired
}

export default TabContentHeader;
