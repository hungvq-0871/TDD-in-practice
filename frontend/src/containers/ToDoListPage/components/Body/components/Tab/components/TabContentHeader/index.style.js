import styled from 'styled-components';

const TabContentHeaderStyle = styled.div`
  ${({ theme }) => (`
    display: flex;
    justify-content: space-between;
    padding: 30px 45px;
    font-weight: bold;

    .close-day {
      color: ${theme.grey1}
    }
  `)}
`;

export default TabContentHeaderStyle;