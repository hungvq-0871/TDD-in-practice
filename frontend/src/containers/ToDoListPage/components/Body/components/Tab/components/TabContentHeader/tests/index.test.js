import React from 'react';
import { render } from '@testing-library/react';

import TabContentHeader from '../index';

const renderTabContentHeader = () => {
  return render(<TabContentHeader dueToday={2} closeToday={3} />);
}

describe(`test how "TabContentHeader" is rendered`, () => {
  test(`the "Due Today(...)" label should display the number of tasks dued today correctly`, () => {
    const { container } = renderTabContentHeader();
    const dueTodayCounterEle = container.querySelector('.due-today-counter');

    expect(dueTodayCounterEle.innerHTML).toBe('2');
  });

  test(`the "Close Today(...)" label should display the number of tasks closed today correctly`, () => {
    const { container } = renderTabContentHeader();
    const dueTodayCounterEle = container.querySelector('.close-today-counter');

    expect(dueTodayCounterEle.innerHTML).toBe('3');
  });

  test(`the "Close Today(...)" label should display the number of tasks closed today correctly`, () => {
    const { container } = renderTabContentHeader();

    expect(container.firstChild).toMatchSnapshot();
  });
});
