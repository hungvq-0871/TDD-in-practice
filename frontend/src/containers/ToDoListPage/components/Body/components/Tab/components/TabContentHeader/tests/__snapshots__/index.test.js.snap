// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test how "TabContentHeader" is rendered the "Close Today(...)" label should display the number of tasks closed today correctly 1`] = `
<div
  class="sc-bdVaJa ePtFUd"
>
  <span>
    Due Today (
    <span
      class="due-today-counter"
    >
      2
    </span>
    )
  </span>
  <span
    class="close-day"
  >
    Close Today (
    <span
      class="close-today-counter"
    >
      3
    </span>
    )
  </span>
</div>
`;
